tests/framework/filters/HostControlTest.php

```diff
@@ -131,4 +131,31 @@ public function testDenyCallback()
         $this->assertFalse($filter->beforeAction($action));
         $this->assertTrue($this->denyCallBackCalled, 'denyCallback should have been called.');
     }
-}
\ No newline at end of file
+
+    public function testDefaultHost()
+    {
+        $filter = new HostControl();
+        $filter->allowedHosts = ['example.com'];
+        $filter->fallbackHostInfo = 'http://yiiframework.com';
+        $filter->denyCallback = function() {};
+
+        $controller = new Controller('test', Yii::$app);
+        $action = new Action('test', $controller);
+        $filter->beforeAction($action);
+
+        $this->assertSame('yiiframework.com', Yii::$app->getRequest()->getHostName());
+    }
+
+    public function testErrorHandlerWithDefaultHost()
+    {
+        $this->setExpectedException('yii\web\NotFoundHttpException', 'Page not found.');
```

Comment:
I'd like to avoid annotations, they are limited and tricky. `setExpectedException()` is already used multiple times in tests, changing this in one place will not make any difference.